mkosi

```diff
@@ -457,6 +457,34 @@ def install_arch(args, workspace, run_build_script):
 
     subprocess.run(cmdline, check=True)
 
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-networkd"], check=True)
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-resolved"], check=True)
+    os.remove(os.path.join(workspace, "root", "etc/resolv.conf"))
+    os.symlink("../usr/lib/systemd/resolv.conf", os.path.join(workspace, "root", "etc/resolv.conf"))
+
+    nsswitch_old = os.path.join(workspace, "root", "etc/nsswitch.conf")
+    nsswitch_new = os.path.join(workspace, "root", "etc/nsswitch.conf.new")
+
+    with open(nsswitch_old, "r") as old:
+        with open(nsswitch_new, "w") as new:
+            for line in old:
+                if line.startswith("hosts:"):
+                    new.write(" ".join(["resolve" if x == "dns" else x for x in line.split(" ")]))
+                else:
+                    new.write(line)
+
+    shutil.copystat(nsswitch_old, nsswitch_new)
+    os.remove(nsswitch_old)
+    shutil.move(nsswitch_new, nsswitch_old)
+
+    if args.bootable:
+        with open(os.path.join(workspace, "root", "etc/systemd/network/wired.network"), "w") as f:
+            f.write("[Match]\n")
+            f.write("Name=en*\n")
+            f.write("\n")
+            f.write("[Network]\n")
+            f.write("DHCP=yes\n")
+
     print_step("Installing ArchLinux complete.")
```

Comment:
I’ve put the creation of the `*.network` config file in the same function as the one that enables networkd (`enable_networkd(…)`).
Would you need to have a dedicated one instead?

Also, for some reasons, using `Type=ether` instead of `Name=en*` doesn’t work on ArchLinux. The interface remains unconfigured:

```
[root@localhost ~]# networkctl list
IDX LINK             TYPE               OPERATIONAL SETUP     
  1 lo               loopback           carrier     unmanaged 
  2 ens3             ether              off         unmanaged 

2 links listed.
```

According to `systemd.network(5)` man page, the `Type=` value has to match the `DEVTYPE` udev property, but I have the feeling that this property is not set:

```
[root@localhost ~]# udevadm info /sys/devices/pci0000\:00/0000\:00\:03.0/net/ens3
P: /devices/pci0000:00/0000:00:03.0/net/ens3
E: DEVPATH=/devices/pci0000:00/0000:00:03.0/net/ens3
E: ID_BUS=pci
E: ID_MODEL_FROM_DATABASE=82540EM Gigabit Ethernet Controller (QEMU Virtual Machine)
E: ID_MODEL_ID=0x100e
E: ID_NET_DRIVER=e1000
E: ID_NET_LINK_FILE=/usr/lib/systemd/network/99-default.link
E: ID_NET_NAME=ens3
E: ID_NET_NAME_MAC=enx525400123456
E: ID_NET_NAME_PATH=enp0s3
E: ID_NET_NAME_SLOT=ens3
E: ID_PATH=pci-0000:00:03.0
E: ID_PATH_TAG=pci-0000_00_03_0
E: ID_PCI_CLASS_FROM_DATABASE=Network controller
E: ID_PCI_SUBCLASS_FROM_DATABASE=Ethernet controller
E: ID_VENDOR_FROM_DATABASE=Intel Corporation
E: ID_VENDOR_ID=0x8086
E: IFINDEX=2
E: INTERFACE=ens3
E: SUBSYSTEM=net
E: SYSTEMD_ALIAS=/sys/subsystem/net/devices/ens3 /sys/subsystem/net/devices/ens3
E: TAGS=:systemd:
E: USEC_INITIALIZED=2057125
```

The systemd version that I used was 231. I haven’t checked the udev rules yet to figure out whether this is a systemd regression or an arch-specific issue.